File: app/src/main/java/com/testndk/jnistudy/ui/activity/BitmapOptionActivity.java
package com.testndk.jnistudy.ui.activity;

import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.widget.ImageView;
import com.testndk.jnistudy.R;

public class BitmapOptionActivity extends BaseActivity {

    @Override
    public int initLayout() {
        return R.layout.activity_bitmap_option;
    }

    @Override
    public void initView() {
        ImageView viewById = findViewById(R.id.ivBitmap);
        BitmapFactory.Options options = new BitmapFactory.Options();
        options.inDensity = 2160;
        options.inScaled = true;
        options.inTargetDensity = 1080;
        options.inPreferredConfig = Bitmap.Config.RGB_565;
        Bitmap bitmap = BitmapFactory.decodeResource(getResources(), R.drawable.test_img, options);

        viewById.setImageBitmap(bitmap);
    }
}

File: app/src/main/java/com/testndk/jnistudy/ui/activity/BitmapOptionActivity.kt
package com.testndk.jnistudy.ui.activity

import android.graphics.Bitmap
import android.graphics.BitmapFactory
import android.widget.ImageView
import com.testndk.jnistudy.R
import com.testndk.jnistudy.ui.BaseActivity

class BitmapOptionActivity : BaseActivity() {
   override fun initLayout() = R.layout.activity_bitmap_option

   override fun initView() {
       val bitmap = BitmapFactory.decodeResource(
           resources,
           R.drawable.test_img,
           BitmapFactory.Options().apply {
               inDensity = 2160
               inScaled = true
               inTargetDensity = 1080
               inPreferredConfig = Bitmap.Config.RGB_565
           })
       findViewById<ImageView>(R.id.ivBitmap).setImageBitmap(bitmap)
   }
}
